File: android/src/main/kotlin/com/dns_technologies/mlkit_scanner/extensions/BarcodeExtension.kt
package com.dns_technologies.mlkit_scanner.extensions

import com.google.mlkit.vision.barcode.common.Barcode

/** Creates json for transmission over the platform channel. */
fun Barcode.toJson(): Map<String, Any?> {
    val mappedFormat = when (format) {
        Barcode.FORMAT_UNKNOWN -> 0
        else -> format
    }

    return mapOf(
            "raw_value" to rawValue,
            "display_value" to displayValue,
            "format" to mappedFormat,
            "value_type" to valueType,
    )
}

File: ios/Classes/Extensions/BarcodeExtension.swift
//
//  BarcodeExtension.swift
//  mlkit_scanner
//
//  Created by ООО "ДНС Технологии" on 12.04.2023.
//

import Foundation
import MLKitBarcodeScanning

extension Barcode {

    /// Maps an accepted recognition result into backend-independent scanner data.
    var scannerBarcode: ScannerBarcode {
        ScannerBarcode(rawValue: rawValue!, displayValue: displayValue,
            format: format == .all ? 0 : format.rawValue, valueType: valueType.rawValue)
    }

    /// Creates json for transmission over the platform channel.
    func toJson() -> [String: Any?] {
        scannerBarcode.toJson()
    }
}
